### PR TITLE
Remove support for Selenium RC

### PR DESCRIPTION
**Pull request by davehunt**

### CODE REVIEW

**Comment by davehunt** [comment]
@bobsilverberg please take a look at this when you get a chance.

File: pytest_mozwebqa/pytest_mozwebqa.py [hunk 1]
```diff
@@ -137,7 +135,7 @@ def pytest_runtest_setup(item):
 
 def pytest_runtest_teardown(item):
     if hasattr(TestSetup, 'selenium') and TestSetup.selenium and 'skip_selenium' not in item.keywords:
```
Comment (bobsilverberg) [inline pytest_mozwebqa/pytest_mozwebqa.py]: I know this is untouched code, but I wonder about this line. Is it possible for `TestSetup.selenium` to be `True` and `skip_selenium` to be present in `item.keywords`? It just seems like the last part of the condition is unnecessary. Wouldn't `TestSetup.selenium` being `True` mean that there is a session to quit?

**Comment by bobsilverberg** [comment]
This looks great to me, @davehunt. I have one question which I added above, plus another I will ask here: it looks like you have removed the `--api` command line option, rather than just deprecating it. Will this break backwards compatibility for anyone who is already using that option to specify `--api=webdriver`? I realize they can just change their command line, but is that something we want to require?

**Comment by bobsilverberg** [comment]
@davehunt I am also seeing one test failure locally on `testShouldErrorThatItCantFindTheChromeBinary`. It looks like the exception message is different than the test expects. Here's the full stack trace:

```
================================================================= FAILURES ==================================================================
_______________________________________________ testShouldErrorThatItCantFindTheChromeBinary ________________________________________________

testdir = <TmpTestdir local('/private/var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/pytest-56/testdir/testShouldErrorThatItCantFindTheChromeBinary0')>
webserver = <webserver.SimpleWebServer object at 0x10f55d890>

    def testShouldErrorThatItCantFindTheChromeBinary(testdir, webserver):
        file_test = testdir.makepyfile("""
            import pytest
            @pytest.mark.nondestructive
            def test_selenium(mozwebqa):
                assert True
        """)
        reprec = testdir.inline_run('--baseurl=http://localhost:%s' % webserver.port,
                                    '--driver=chrome',
                                    '--chromeopts={"binary_location":"foo"}',
                                    file_test)
        passed, skipped, failed = reprec.listoutcomes()
        assert len(failed) == 1
        out = failed[0].longrepr.reprcrash.message
        if 'ChromeDriver executable needs to be available in the path' in out:
            pytest.fail('You must have Chrome Driver installed on your path for this test to run correctly. '
                        'For further information see pytest-mozwebqa documentation.')
>       assert 'Could not find Chrome binary at: foo' in out
E       assert 'Could not find Chrome binary at: foo' in 'WebDriverException: Message: unknown error: no chrome binary at foo\n  (Driver info: chromedriver=2.9.248307,platform=Mac OS X 10.10.2 x86_64)'

/Users/bsilverberg/gitRepos/pytest-mozwebqa/testing/test_usage.py:162: AssertionError
----------------------------------------------------------- Captured stdout call ------------------------------------------------------------
============================= test session starts ==============================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.7.0
rootdir: /private/var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/pytest-56/testdir/testShouldErrorThatItCantFindTheChromeBinary0, inifile:
plugins: mozwebqa
collected 1 items

testShouldErrorThatItCantFindTheChromeBinary.py E

==================================== ERRORS ====================================
_______________________ ERROR at setup of test_selenium ________________________

item = <Function 'test_selenium'>

    def pytest_runtest_setup(item):
        item.debug = {
            'urls': [],
            'screenshots': [],
            'html': []}
        TestSetup.base_url = item.config.option.base_url

        # configure test proxies
        if hasattr(item.config, 'browsermob_test_proxy'):
            item.config.option.proxy_host = item.config.option.bmp_host
            item.config.option.proxy_port = item.config.browsermob_test_proxy.port

        # consider this environment sensitive if the base url or any redirection
        # history matches the regular expression
        sensitive = False
        if TestSetup.base_url and not item.config.option.skip_url_check:
            r = requests.get(TestSetup.base_url, verify=False, timeout=REQUESTS_TIMEOUT)
            urls = [h.url for h in r.history] + [r.url]
            matches = [re.search(item.config.option.sensitive_url, u) for u in urls]
            sensitive = any(matches)

        destructive = 'nondestructive' not in item.keywords

        if (sensitive and destructive):
            first_match = matches[next(i for i, match in enumerate(matches) if match)]

            # skip the test with an appropriate message
            py.test.skip('This test is destructive and the target URL is '
                         'considered a sensitive environment. If this test is '
                         'not destructive, add the \'nondestructive\' marker to '
                         'it. Sensitive URL: %s' % first_match.string)

        if item.config.option.sauce_labs_credentials_file:
            item.sauce_labs_credentials = credentials.read(item.config.option.sauce_labs_credentials_file)

        if item.config.option.credentials_file:
            TestSetup.credentials = credentials.read(item.config.option.credentials_file)

        test_id = '.'.join(split_class_and_test_names(item.nodeid))

        if 'skip_selenium' not in item.keywords:
            if hasattr(item, 'sauce_labs_credentials'):
                from sauce_labs import Client
                TestSetup.selenium_client = Client(
                    test_id,
                    item.config.option,
                    item.keywords,
                    item.sauce_labs_credentials)
            else:
                from selenium_client import Client
                TestSetup.selenium_client = Client(
                    test_id,
                    item.config.option)
>           TestSetup.selenium_client.start()

/Users/bsilverberg/gitRepos/pytest-mozwebqa/pytest_mozwebqa/pytest_mozwebqa.py:126:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/bsilverberg/gitRepos/pytest-mozwebqa/pytest_mozwebqa/selenium_client.py:66: in start
    self.start_client()
/Users/bsilverberg/gitRepos/pytest-mozwebqa/pytest_mozwebqa/selenium_client.py:122: in start_client
    desired_capabilities=capabilities or None)
/Users/bsilverberg/.virtualenvs/pytest-mozwebqa/lib/python2.7/site-packages/selenium-2.45.0-py2.7.egg/selenium/webdriver/chrome/webdriver.py:65: in __init__
    keep_alive=True)
/Users/bsilverberg/.virtualenvs/pytest-mozwebqa/lib/python2.7/site-packages/selenium-2.45.0-py2.7.egg/selenium/webdriver/remote/webdriver.py:74: in __init__
    self.start_session(desired_capabilities, browser_profile)
/Users/bsilverberg/.virtualenvs/pytest-mozwebqa/lib/python2.7/site-packages/selenium-2.45.0-py2.7.egg/selenium/webdriver/remote/webdriver.py:123: in start_session
    'desiredCapabilities': desired_capabilities,
/Users/bsilverberg/.virtualenvs/pytest-mozwebqa/lib/python2.7/site-packages/selenium-2.45.0-py2.7.egg/selenium/webdriver/remote/webdriver.py:175: in execute
    self.error_handler.check_response(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x110d45510>
response = {'sessionId': '5384e0b95648ebb8b975205f49c673a1', 'status': 13, 'value': {'message': 'unknown error: no chrome binary at foo
  (Driver info: chromedriver=2.9.248307,platform=Mac OS X 10.10.2 x86_64)'}}

    def check_response(self, response):
        """
            Checks that a JSON response from the WebDriver does not have an error.

            :Args:
             - response - The JSON response from the WebDriver server as a dictionary
               object.

            :Raises: If the response contains an error message.
            """
        status = response['status']
        if status == ErrorCode.SUCCESS:
            return
        exception_class = ErrorInResponseException
        if status in ErrorCode.NO_SUCH_ELEMENT:
            exception_class = NoSuchElementException
        elif status in ErrorCode.NO_SUCH_FRAME:
            exception_class = NoSuchFrameException
        elif status in ErrorCode.NO_SUCH_WINDOW:
            exception_class = NoSuchWindowException
        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:
            exception_class = StaleElementReferenceException
        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:
            exception_class = ElementNotVisibleException
        elif status in ErrorCode.INVALID_ELEMENT_STATE:
            exception_class = InvalidElementStateException
        elif status in ErrorCode.INVALID_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER:
            exception_class = InvalidSelectorException
        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:
            exception_class = ElementNotSelectableException
        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:
            exception_class = WebDriverException
        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:
            exception_class = WebDriverException
        elif status in ErrorCode.TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.SCRIPT_TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.UNKNOWN_ERROR:
            exception_class = WebDriverException
        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:
            exception_class = UnexpectedAlertPresentException
        elif status in ErrorCode.NO_ALERT_OPEN:
            exception_class = NoAlertPresentException
        elif status in ErrorCode.IME_NOT_AVAILABLE:
            exception_class = ImeNotAvailableException
        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:
            exception_class = ImeActivationFailedException
        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:
            exception_class = MoveTargetOutOfBoundsException
        else:
            exception_class = WebDriverException
        value = response['value']
        if isinstance(value, basestring):
            if exception_class == ErrorInResponseException:
                raise exception_class(response, value)
            raise exception_class(value)
        message = ''
        if 'message' in value:
            message = value['message']

        screen = None
        if 'screen' in value:
            screen = value['screen']

        stacktrace = None
        if 'stackTrace' in value and value['stackTrace']:
            stacktrace = []
            try:
                for frame in value['stackTrace']:
                    line = self._value_or_default(frame, 'lineNumber', '')
                    file = self._value_or_default(frame, 'fileName', '<anonymous>')
                    if line:
                        file = "%s:%s" % (file, line)
                    meth = self._value_or_default(frame, 'methodName', '<anonymous>')
                    if 'className' in frame:
                        meth = "%s.%s" % (frame['className'], meth)
                    msg = "    at %s (%s)"
                    msg = msg % (meth, file)
                    stacktrace.append(msg)
            except TypeError:
                pass
        if exception_class == ErrorInResponseException:
            raise exception_class(response, message)
        elif exception_class == UnexpectedAlertPresentException and 'alert' in value:
            raise exception_class(message, screen, stacktrace, value['alert'].get('text'))
>       raise exception_class(message, screen, stacktrace)
E       WebDriverException: Message: unknown error: no chrome binary at foo
E         (Driver info: chromedriver=2.9.248307,platform=Mac OS X 10.10.2 x86_64)

/Users/bsilverberg/.virtualenvs/pytest-mozwebqa/lib/python2.7/site-packages/selenium-2.45.0-py2.7.egg/selenium/webdriver/remote/errorhandler.py:166: WebDriverException
=========================== 1 error in 1.10 seconds ============================
```

File: pytest_mozwebqa/pytest_mozwebqa.py (hunk 1 above)
Comment (davehunt) [inline pytest_mozwebqa/pytest_mozwebqa.py]: Possibly, but I don't want to change this here. Please feel free to write a patch and run some tests.

**Comment by davehunt** [comment]
pytest-mozwebqa 2.0 will break backwards compatibility. It's likely there will be more breakages to come. It's hard to know how many people might be affected, but we're certainly going to break stuff in the next major version. Any users affected could just pin to an older version.

**Comment by davehunt** [comment]
The test failure does not concern me - it's likely a path issue or missing chrome driver. It's passing on Travis, and is unrelated to the changes. I will likely revisit this area during the 2.0 work.

**Pull request merged**